samples/matter/common/src/bt_nus_service.cpp

```diff
@@ -0,0 +1,224 @@
+	mAdvertisingItems[1] = BT_DATA(BT_DATA_NAME_COMPLETE, name, static_cast<uint8_t>(nameLen));
+
+	mServiceItems[0] = BT_DATA(BT_DATA_UUID128_ALL, BT_UUID_NUS_VAL, sizeof(BT_UUID_NUS_VAL));
+
+	mAdvertisingRequest.priority = priority;
+	mAdvertisingRequest.options = kAdvertisingOptions;
+	mAdvertisingRequest.minInterval = minInterval;
+	mAdvertisingRequest.maxInterval = maxInterval;
+	mAdvertisingRequest.advertisingData = Span<bt_data>(mAdvertisingItems);
+	mAdvertisingRequest.scanResponseData = Span<bt_data>(mServiceItems);
+
+	mAdvertisingRequest.onStarted = [](int rc) {
+		if (rc == 0) {
+			ChipLogDetail(DeviceLayer, "NUS BLE advertising started");
+		} else {
+			ChipLogError(DeviceLayer, "Failed to start NUS BLE advertising: %d", rc);
+		}
+	};
+	mAdvertisingRequest.onStopped = []() { ChipLogDetail(DeviceLayer, "NUS BLE advertising stopped"); };
+
+	settings_load();
+#if (CONFIG_BT_FIXED_PASSKEY)
+	VerifyOrReturnError(bt_passkey_set(kDefaultPasskey) != 0, CHIP_ERROR_INTERNAL);
+#endif
+	VerifyOrReturnError(bt_conn_auth_cb_register(&sConnAuthCallbacks) == 0, CHIP_ERROR_INTERNAL);
+	VerifyOrReturnError(bt_conn_auth_info_cb_register(&sConnAuthInfoCallbacks) == 0, CHIP_ERROR_INTERNAL);
+	VerifyOrReturnError(bt_nus_init(&sNusCallbacks) == 0, CHIP_ERROR_INTERNAL);
+
+	return CHIP_NO_ERROR;
+}
+
+void NUSService::StartServer()
+{
+	VerifyOrReturn(!mIsStarted, ChipLogError(DeviceLayer, "NUS service was already started"));
+
+	PlatformMgr().LockChipStack();
+	CHIP_ERROR ret = BLEAdvertisingArbiter::InsertRequest(mAdvertisingRequest);
+	PlatformMgr().UnlockChipStack();
+
+	if (CHIP_NO_ERROR != ret) {
+		ChipLogError(DeviceLayer, "Could not start NUS service");
+		mIsStarted = false;
+		return;
+	}
+	mIsStarted = true;
+	ChipLogProgress(DeviceLayer, "NUS service started");
+}
+
+void NUSService::StopServer()
+{
+	VerifyOrReturn(mIsStarted);
+
+	PlatformMgr().LockChipStack();
+	BLEAdvertisingArbiter::CancelRequest(mAdvertisingRequest);
+	PlatformMgr().UnlockChipStack();
+
+	mIsStarted = false;
+	ChipLogProgress(DeviceLayer, "NUS service stopped");
+}
+
+void NUSService::RxCallback(struct bt_conn *conn, const uint8_t *const data, uint16_t len)
+{
+	VerifyOrReturn(bt_conn_get_security(GetNUSService().mBTConnection) >= BT_SECURITY_L2);
+	ChipLogDetail(DeviceLayer, "NUS received: %d bytes", len);
+	GetNUSService().DispatchCommand(reinterpret_cast<const char *>(data), len);
+}
+
+CHIP_ERROR NUSService::RegisterCommand(const char *const name, size_t length, CommandCallback callback, void *context)
+{
+	VerifyOrReturnError(name, CHIP_ERROR_INVALID_ARGUMENT);
+
+	struct Command newCommand {};
+	memcpy(newCommand.command, name, length);
+	newCommand.commandLen = length;
+	newCommand.callback = callback;
+	newCommand.context = context;
+	mCommandsList.push_back(newCommand);
+
+	return CHIP_NO_ERROR;
+}
+
+void NUSService::DispatchCommand(const char *const data, uint16_t len)
+{
+	for (auto c : mCommandsList) {
+		if (strncmp(data, c.command, len) == 0) {
```

Comment:
Shouldn't we also check the size? Or is `data` null-terminated? Otherwise, if one put "Lo" it would match the "Lock" command. Not sure if that's a bug or a feature :)